services/user/src/middleware/basic-access-control.ts

```diff
@@ -0,0 +1,66 @@
+import jwt from 'jsonwebtoken';
+import { findUserById as _findUserById } from '../model/repository';
+import { NextFunction, Request, Response } from 'express';
+import { handleForbidden, handleInternalError, handleUnauthorized } from '../utils/helper';
+
+export function verifyAccessToken(req: Request, res: Response, next: NextFunction) {
+    const authHeader = req.headers['authorization'];
+    if (!authHeader) {
+        handleUnauthorized(res, 'Authenticated failed');
+        return;
+    }
+
+    // request auth header: `Authorization: Bearer + <access_token>`
+    const token = authHeader.split(' ')[1];
+
+    if (!process.env.JWT_SECRET) {
+        handleInternalError(res, 'JWT Secret not provided');
+        return;
+    }
+
```

Comment:
When I first read this I was wondering if we need to verify if ```authHeader.split(' ')[0] === "Bearer"```. Functionally its going to be the same, but logically a bit wrong.